lib/cc/engine/analyzers/violation.rb

```diff
@@ -69,23 +69,25 @@ def format_other_locations
         end
 
         def format_sexp(sexp)
+          lines = sexp_lines(sexp)
           {
             "path": sexp.file.gsub(%r(^./), ""),
             "lines": {
-              "begin": sexp.line,
-              "end": sexp.end_line || sexp_max_line(sexp, sexp.line)
+              "begin": lines[0],
+              "end": lines[1]
             }
           }
         end
 
-        def sexp_max_line(sexp_tree, default)
-          max = default
+        def sexp_lines(sexp_tree)
```

Comment:
This method is getting a little fiddly -- what do you think about a small class? The class could then have methods for the begin/end so you don't need the array return value either.